Test/AwsCommonRuntimeKitTests/sdkutils/CRTAWSEndpointsRuleEngineTests.swift

```diff
@@ -0,0 +1,210 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import XCTest
+import Foundation
+@testable import AwsCommonRuntimeKit
+
+class CRTAWSEndpointsRuleEngineTests: CrtXCBaseTestCase {
```

Comment:
We should be using tracing allocator in tests to detect memory leaks. However, this will fail the tests because currently Swift CRT has a lot of leaks.

```suggestion
//TODO: use tracing allocator
class CRTAWSEndpointsRuleEngineTests: CrtXCBaseTestCase {
```

Reply:
We are already using 
https://github.com/awslabs/aws-crt-swift/blob/main/Test/AwsCommonRuntimeKitTests/CrtXCBaseTestCase.swift#L8-L9